source/extensions/omni.isaac.lab/omni/isaac/lab/sim/spawners/wrappers/wrappers.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) 2022-2024, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import random
+import re
+from typing import TYPE_CHECKING
+
+import omni.isaac.core.utils.prims as prim_utils
+import omni.usd
+from pxr import Sdf, Usd
+
+import omni.isaac.lab.sim as sim_utils
+from omni.isaac.lab.sim.spawners.from_files import UsdFileCfg
+
+if TYPE_CHECKING:
+    from . import wrappers_cfg
+
+
+def spawn_multi_asset(
+    prim_path: str,
+    cfg: wrappers_cfg.MultiAssetSpawnerCfg,
+    translation: tuple[float, float, float] | None = None,
+    orientation: tuple[float, float, float, float] | None = None,
+) -> Usd.Prim:
+    """Spawn multiple assets based on the provided configurations.
+
+    This function spawns multiple assets based on the provided configurations. The assets are spawned
+    in the order they are provided in the list. If the `random_choice` parameter is set to True, a random
+    asset configuration is selected for each spawn.
+
+    Args:
+        prim_path: The prim path to spawn the assets.
+        cfg: The configuration for spawning the assets.
+        translation: The translation of the spawned assets. Default is None.
+        orientation: The orientation of the spawned assets. Default is None.
+
+    Returns:
+        The created prim at the first prim path.
+    """
+    # resolve: {SPAWN_NS}/AssetName
+    # note: this assumes that the spawn namespace already exists in the stage
+    root_path, asset_path = prim_path.rsplit("/", 1)
+    # check if input is a regex expression
+    # note: a valid prim path can only contain alphanumeric characters, underscores, and forward slashes
+    is_regex_expression = re.match(r"^[a-zA-Z0-9/_]+$", root_path) is None
+
+    # resolve matching prims for source prim path expression
+    if is_regex_expression and root_path != "":
+        source_prim_paths = sim_utils.find_matching_prim_paths(root_path)
+        # if no matching prims are found, raise an error
+        if len(source_prim_paths) == 0:
+            raise RuntimeError(
+                f"Unable to find source prim path: '{root_path}'. Please create the prim before spawning."
+            )
+    else:
+        source_prim_paths = [root_path]
+
+    # spawn everything first in a "Dataset" prim
+    prim_utils.create_prim("/World/Dataset", "Scope")
```

Comment:
Maybe make this a default value in the config? Just in case that someone has a /World/Dataset prim defined, for whatever reason.

Reply:
Good point!

Reply:
I added a check to get the next available free path to be always sure.